table/iterator.go

```diff
@@ -204,10 +204,30 @@ func (itr *Iterator) seekHelper(blockIdx int, key []byte) {
 		return
 	}
 	itr.bi.setBlock(block)
-	itr.bi.seek(key)
+	itr.bi.seek(key, 0)
 	itr.err = itr.bi.Error()
 }
 
+const seekBound = 3
+
+func (itr *Iterator) seekFromOffset(blockIdx int, offset int, key []byte) {
+	itr.bpos = blockIdx
+	block, err := itr.t.block(blockIdx)
+	if err != nil {
+		itr.err = err
+		return
+	}
+	itr.bi.setBlock(block)
+	for i := 0; i < seekBound; i++ {
+		itr.bi.setIdx(offset)
+		if y.CompareKeys(itr.bi.key, key) >= 0 {
+			return
+		}
+		offset++
+	}
+	itr.bi.seek(key, offset)
```

Comment:
Is it possible that the lookup key has the same hash value as the key in the table, the blockIdx and offset is wrong?

Reply:
* If the key exist in table, and a hash collision will be returned by `PointGet`. Then fallback to normal seek.
* If the key doesn't exist in table, and a hash collision will cause returned iterator point to a wrong position, caller need to check the `Iterator.Key()` use `y.SameKey()`.

Reply:
It would be simpler and safer if we check the key is the same here or in `PointGet`.

Reply:
This logical already in `(*levelHandler).get(key []byte)` method, because `normal seek search` also need do this check.

Reply:
Got it. I'm going to do a little refactor in `(*levelHandler).get(key []byte)`.